src/shogun/clustering/KMeansBase.cpp

```diff
@@ -317,19 +317,25 @@ SGMatrix<float64_t> CKMeansBase::kmeanspp()
 
 void CKMeansBase::init()
 {
-	max_iter=300;
-	k=8;
-	dimensions=0;
-	fixed_centers=false;
-	use_kmeanspp=false;
-	SG_ADD(&max_iter, "max_iter", "Maximum number of iterations", ParameterProperties::HYPER);
-	SG_ADD(&k, "k", "k, the number of clusters", ParameterProperties::HYPER);
+	max_iter = 300;
+	k = 8;
+	dimensions = 0;
+	fixed_centers = false;
+	use_kmeanspp = false;
+	SG_ADD(
+	    &max_iter, "max_iter", "Maximum number of iterations",
+	    ParameterProperties::HYPER);
+	SG_ADD(
+	    &k, "k", "k, the number of clusters",
+	    ParameterProperties::HYPER | ParameterProperties::CONSTRAIN,
+	    SG_CONSTRAINT(positive<>()));
```

Comment:
the `<>` has to stay because gcc cannot do the deduction without it until gcc-9...